lib/ui/screens/result/result_screen.dart

```diff
@@ -66,16 +68,26 @@ class ResultContent extends StatelessWidget {
                                 AppText.body(state.description!),
                             ]),
                       ),
-                      Button.outlined(
-                        text: localization.confirmButtonTitle,
-                        onPressed: () {
-                          final event = NavigateToHomeEvent(
-                            resultType: state.type,
-                            operationType: state.operationType,
-                          );
-                          context.read<ResultBloc>().add(event);
-                        },
-                      ),
+                      if (!Platform.isIOS ||
```

Comment:
FYI: This is the new part you should review. @tamas-toth, @balazs-gerlei.

Reply:
🔧 I think understanding this condition, also alongside the other one inside the onPressed event handler logic is unnecessarily difficult, I would really appreciate simplifying it to clearly communicate intention.

If I understood correctly, this button is always shown on Android, and on iOS it's only shown when the error is not fatal. And when the button is pressed, on Android the app is closed when the error is a fatal one, otherwise the app navigates back to the home screen (which is always the case on iOS).

If I get that right, I would recommend the following changes:

1. Change the first condition to `Platform.isAndroid || (Platform.isIOS && state.type != OperationResultType.fatal)`
2. Change the condition inside the event handler to `state.type == OperationResultType.fatal`

WDYT @laszlo-domonkos (and @tamas-toth)?

Reply:
I'm fine with your proposal Balázs.
Btw in the second case checking the platform does not make sense at this moment as well.

Reply:
I fixed it and I added more inline documentation.